opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -274,8 +266,8 @@ impl opentelemetry::logs::Logger for Logger {
         }
 
         let mut data = LogData {
-            record: log_record,
-            instrumentation: self.instrumentation_library().clone(),
+            record: Cow::Borrowed(&log_record),
```

Comment:
thinking out loud -- what is the need of a separate `LogData` struct itself? Can we use `LogRecord` itself?

LogRecord struct can add ref (not Arc else we'd need to touch heap again!) to InstrumentationLibrary. In create_log_record() method, the Logger can set the InstrumentationLibrary into the `LogRecord` right at creation time. With this, sdk does not need to create another structure (LogData), and instead can pass `LogRecord` as-is.

(I haven't thought through 'a issues, if any)

Reply:
Need to do more thinking around this. First thoughts are - If LogRecord directly holds a reference to InstrumentationLibrary, we need to carefully manage the lifetimes of these references. This could complicate the API and usage patterns, particularly if InstrumentationLibrary has a shorter lifetime than the LogRecord (eg logger gets destroyed while the LogData is in queue).  Doable, but unless I am overthinking, it seems complex :)

Reply:
If any component (Batchprocessor) needs to queue it/store beyond the 'a, it should clone and store. Just like it does today.

Reply:
Yes, but cloning/storing would be difficult with LogRecording containing `a reference. As of now, things are much simpler as LogRecord either has owning data, or Cow.